src/main.ts

```diff
@@ -17,6 +17,46 @@ import {ContextInfo} from '@docker/actions-toolkit/lib/types/docker/docker';
 import * as context from './context';
 import * as stateHelper from './state-helper';
 
+/**
+ * Retry a function with exponential backoff
+ */
+async function retryWithBackoff<T>(
+  operation: () => Promise<T>,
+  maxRetries: number = 3,
+  initialDelay: number = 1000,
+  maxDelay: number = 10000,
+  shouldRetry: (error: Error) => boolean = () => true
+): Promise<T> {
+  let retries = 0;
+  let delay = initialDelay;
+  
+  while (true) {
+    try {
+      return await operation();
+    } catch (error) {
+      if (retries >= maxRetries || !shouldRetry(error)) {
+        throw error;
+      }
+      
+      retries++;
+      core.info(`Retry ${retries}/${maxRetries} after ${delay}ms due to: ${error.message}`);
+      await new Promise(resolve => setTimeout(resolve, delay));
+      
+      // Exponential backoff with jitter
+      delay = Math.min(delay * 2, maxDelay) * (0.8 + Math.random() * 0.4);
+    }
+  }
+}
+
+/**
+ * Check if an error is a buildkit socket connection error
+ */
+function isBuildkitSocketError(error: Error): boolean {
+  return error.message.includes('/run/buildkit/buildkitd.sock') || 
+         error.message.includes('failed to list workers') ||
+         error.message.includes('connection error');
+}
```

Comment:
Not sure we should check for this kind of error. We should just simply retry if the command fails.